### PR TITLE
Add test for feature RPC (RIPD-1391):

### PR DESCRIPTION
**Pull request by mellery451**
Create unit test for feature RPC method. Add client_error field to env
RPC requests to provide information about parsing errors.

### CODE REVIEW

**Comment by mellery451** [comment]
the drop on coverage is because sweepTimer went off (unexpectedly) in 1ede097. I'll be dealing with that in a separate PR.

File: src/test/rpc/Feature_test.cpp [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+    testInvalidFeature()
+    {
+        testcase ("Invalid Feature");
+
+        using namespace test::jtx;
+        Env env {*this};
+
+        auto jrr = env.rpc("feature", "AllTheThings") [jss::result];
+        BEAST_EXPECT(jrr[jss::error] == "badFeature");
+        BEAST_EXPECT(jrr[jss::error_message] == "Feature unknown or invalid.");
```
Comment (nbougalis) [inline src/test/rpc/Feature_test.cpp]: Maybe add one more test here: feature names are case-sensitive, so we could ensure that, say, `CryptoConditions` and `cryptoconditions` don't match up.

Comment (mellery451) [inline src/test/rpc/Feature_test.cpp]: good idea - added.

File: src/test/rpc/Feature_test.cpp [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+                feature[jss::name].asString() + " vote");
+        }
+
+        auto majorities = getMajorityAmendments (*env.closed());
+        if(! BEAST_EXPECT(majorities.empty()))
+            return;
+
+        // close ledgers until the amendments show up.
+        int i = 0;
+        for (i = 0; i <= 256; ++i)
```
Comment (miguelportilla) [inline src/test/rpc/Feature_test.cpp]: `i` is being initialized twice. Suggestion:
`for (auto i = 0; i < 256; ++i)`

File: src/test/rpc/Feature_test.cpp [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+
+    void
+    testSingleFeature()
+    {
+        testcase ("Feature Param");
+
+        using namespace test::jtx;
+        Env env {*this};
+
+        auto jrr = env.rpc("feature", "CryptoConditions") [jss::result];
```
Comment (miguelportilla) [inline src/test/rpc/Feature_test.cpp]: Suggestion:
`BEAST_EXPECTS(jrr[jss::status] == jss::success, "status");`

File: src/test/rpc/Feature_test.cpp [hunk 4]
```diff
@@ -0,0 +1,274 @@
...
+        if(! BEAST_EXPECT(jrr.isMember(jss::features)))
+            return;
+        for(auto it = jrr[jss::features].begin();
+            it != jrr[jss::features].end(); ++it)
+        {
+            uint256 id;
+            id.SetHexExact(it.key().asString().c_str());
+            bool shouldBeEnabled = env.app().getAmendmentTable().isEnabled(id);
+            BEAST_EXPECTS((*it)[jss::enabled].asBool() == shouldBeEnabled,
+                (*it)[jss::name].asString() + " enabled");
```
Comment (miguelportilla) [inline src/test/rpc/Feature_test.cpp]: Is it worth checking supported?
`            bool supported = env.app().getAmendmentTable().isSupported(id);
            BEAST_EXPECTS((*it)[jss::supported].asBool() == supported,
                (*it)[jss::name].asString() + " supported");`

File: src/test/rpc/Feature_test.cpp [hunk 5]
```diff
@@ -0,0 +1,274 @@
...
+
+        using namespace test::jtx;
+        Env env {*this};
+
+        auto jrr = env.rpc("feature") [jss::result];
+        if(! BEAST_EXPECT(jrr.isMember(jss::features)))
+            return;
+        for(auto const& feature : jrr[jss::features])
+        {
+            //default config - so all should be disabled, not vetoed, and supported
```
Comment (miguelportilla) [inline src/test/rpc/Feature_test.cpp]: The existence of `jss::name` in the result should probably be checked at least once in the unit test.
`if(! BEAST_EXPECT(feature.isMember(jss::name)))
    return;`

File: src/test/rpc/Feature_test.cpp [hunk 6]
```diff
@@ -0,0 +1,274 @@
...
+        testcase ("Veto");
+
+        using namespace test::jtx;
+        Env env {*this,
+            features(featureCryptoConditions)};
+        // The amendment table has to be modified
+        // since that is what feature RPC actually checks
+        env.app().getAmendmentTable().enable(featureCryptoConditions);
+
+        auto jrr = env.rpc("feature", "CryptoConditions") [jss::result];
```
Comment (miguelportilla) [inline src/test/rpc/Feature_test.cpp]: Suggestion:
`BEAST_EXPECTS(jrr[jss::status] == jss::success, "status");`
or
`if (! BEAST_EXPECTS(jrr[jss::status] == jss::success, "status")
    return;`

**Comment by mellery451** [comment]
@miguelportilla thanks for the review - I have addressed your comments in 752098f

**Comment by mellery451** [comment]
this test involves a validator config and will need to be modified following upcoming validator manifest changes

**Pull request closed without merging**